### ISSUE TITLE
AMP Structured Data errors for `publisher.logo` and `article.image`

### ISSUE THREAD

**Issue by AileenCGN** [TryGhost/Ghost#7558]
As with #7557, #7425 and #7396 we're facing more and more issues regarding structured data in AMP. Google wants to have a proper `publisher.logo` (warning) in 60x600px dimensions _and_ an `article.image` (error). Always!

But it's not only for AMP! We get the exact same errors and warnings for normal blog posts, so we need a _short term solution for LTS_, which could be like this:
- replace the default Ghost blog logo with one, that fits the dimensions
- put a text underneath the blog logo update in general settings, that it has to fit the image dimensions
- use the blog cover as a default `article.image` whenever no post image is supplied. If there's no blog cover, use a default picture (this needs to be defined, maybe something neutral, that refers to writing?)

For a _long term solution_, we need image-optimisation. With that we're able to resize the `publisher.logo` and offer the possibility to upload a default post image, just like we do with the blog cover or logo in general settings.

Happy for other opinions here!

**Comment by AileenCGN** [comment]
Google updated the guidelines and made it more clear, which requirements are for AMP specific and which once aren't: https://developers.google.com/search/docs/data-types/articles#logo-guidelines

**Comment by ErisDS** [comment]
@AileenCGN can you post an update on here with what's left to do or otherwise close this issue?

**Comment by AileenCGN** [comment]
Google has the requirements specified for AMP and non-AMP pages now (see [here](https://developers.google.com/search/docs/data-types/articles)). Regarding the two properties `article.image` and `publisher.logo` it looks like this now:

- `article.image`: **required (AMP)**, **recommended (non-AMP)**
   If a post doesn't have a feature image, it'll error especially for AMP. That means, that we would **always** need to supply an image for AMP posts. How can we do this? 
   
   > use the blog cover as a default article.image whenever no post image is supplied. If there's no blog cover, use a default picture (this needs to be defined, maybe something neutral, that refers to writing?)
   
   Besides ^this idea, we could also crawl the content of the post for any image in there and use the first one we find.
   Even though the [Structured Data Testing Tool](https://search.google.com/structured-data/testing-tool/u/0/) renders a warning for non-existent article image on non-AMP posts, it should be fine - according to Google - for those cases.

- `publisher.logo`: **required (AMP)**, **ignored (non-AMP)**
   This changed. It used to be required for non-AMP pages as well, which takes a bit of the pain away, especially since with Ghost 1.0 our default logo is an `.svg` which as particularly not allowed according to [Google](https://developers.google.com/search/docs/data-types/articles#logo-guidelines). Again, the Structured Data Testing Tool shows a warning for both, AMP and non-AMP, as it doesn't really seem to differentiate those cases.

**TL;DR**: This issue might as well be renamed, as it's now **only valid for AMP** posts. Normal blog posts shouldn't be affected by this. I still think that this statement

> For a long term solution, we need image-optimisation. With that we're able to resize the publisher.logo and offer the possibility to upload a default post image, just like we do with the blog cover or logo in general settings.

is still valid, but we might as well have it as a subtask in a possible upcoming image-optimisation issue.

Another quick fix, we could do, is to offer some additional text on the App - AMP page, where the importance of an post feature image and the requirements for the logo are either explained, or linked to (as they might change again).

**Comment by ZsharE** [comment]
Any fix for missing article.image ?

**Comment by ThomasArdal** [comment]
A fallback to the blog cover or something sounds like an easy fix. Currently, Search Console sends warning emails about this issue. And while it may not be an issue in the search results, ignoring Search Console emails can be very dangerous. At some point, they send something critical and you have gotten used to ignoring it 😄 

Could be a two step rocket, where you'd default to the blog cover in v1. In v2, you could implement a custom default.

**Comment by xmsec** [comment]
in the file './versions/1.22.3/core/server/helpers/ghost_head.js', i added this
			if(!metaData.schema.hasOwnProperty('image')){
				metaData.schema['image']={}
			}
and it turned out to be
                    if (metaData.schema) {
			if(!metaData.schema.hasOwnProperty('image')){
				metaData.schema['image']={}
			}
                        head.push('<script type="application/ld+json">\n' +
                            JSON.stringify(metaData.schema, null, '    ') +
                            '\n    </script>\n');
                    }
                }
Then passed the Google Accelerated Mobile Pages(AMP) Validation.
(Maybe it is not a good choice. But i have no idea to solve this problem.)

**Comment by xmsec** [comment]
in the file './versions/1.22.3/core/server/helpers/ghost_head.js', i added this
```
if(!metaData.schema.hasOwnProperty('image')){
	metaData.schema['image']={}
	}
```
and it turned out to be
```
if (metaData.schema) {
	if(!metaData.schema.hasOwnProperty('image')){
		metaData.schema['image']={}
	}
head.push('<script type="application/ld+json">\n' +
JSON.stringify(metaData.schema, null, ' ') +
'\n </script>\n');
}
```
Then passed the Google Accelerated Mobile Pages(AMP) Validation.
(Maybe it is not a good choice. But i have no idea to solve this problem.)

**Comment by ubuntutest** [comment]
Hello,
Any news about "A fallback to the blog cover or something" to fix AMP warnings?

Thank you

**Comment by kirrg001** [comment]
@AileenCGN ping

**Comment by ubuntutest** [comment]
Does anyone know how to do ? :)

**Comment by joshin16** [comment]
I too seeing increase in AMp issues on my portal https://classifieds.meramaal.com do i need to update the plugin please suggest the version..few versions disturbed the functionality

**Comment by ErisDS** [comment]
I recommend that we introduce the concept of "system" image sizes, which have one or two sizes used for schema.org, and where people have linked to external images (no longer possible), we should assume they are the right size.

This would allow us to remove all of the code that handles looking up image dimensions, seeing as we are now in control of what dimensions are served, and simplify the logic extensively.

**Comment by ThomasArdal** [comment]
There is activity now :)

**Comment by aleksandar-todorovic** [comment]
Pardon me asking, but why was this closed? I'm running the latest version (v3.0.2) and I still see the following markup:

```json
    "publisher": {
        "@type": "Organization",
        "name": "Example",
        "logo": "/path/to/logo.png"
    },
```

...instead of:

```
    "publisher": {
        "@type": "Organization",
        "name": "Example",
        "logo": {
          "@type": "ImageObject",
          "url": "/path/to/logo.png",
    },
```

Interestingly, I can successfully run [Structured Data Testing Tool](https://search.google.com/structured-data/testing-tool) on my homepage with no issues nor warnings, but when I do the same on an article (so, a publisher within `"@type": "article")`, I get an error around publisher logo: `A value for the url field is required.`

**Comment by sammccord** [comment]
bumpin'

**Comment by Otoris** [comment]
Bumping again

**Comment by JohnONolan** [comment]
The issue got closed because it has no traction. PRs are welcome. "bumping" comments don't do anything.
